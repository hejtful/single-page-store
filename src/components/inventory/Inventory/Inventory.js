import React from 'react';

import { InventoryItem } from '../InventoryItem';
import { InventoryItemNew } from '../InventoryItemNew';

export const Inventory = ({
  items,
  onInventoryItemChange,
  onInventoryItemRemove,
  onNewInventoryItemSubmit,
}) => (
  <div data-cy="inventory">
    {items?.map((item) => (
      <InventoryItem
        {...item}
        key={item.id}
        onChange={onInventoryItemChange}
        onRemove={onInventoryItemRemove}
      />
    ))}

    <InventoryItemNew onSubmit={onNewInventoryItemSubmit} />
  </div>
);
