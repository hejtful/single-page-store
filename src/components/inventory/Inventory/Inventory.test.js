import React from 'react';
import { render } from '@testing-library/react';

import { Inventory } from './Inventory';

const initialInventoryItem = {
  id: '1',
  title: 'Test product title',
  price: 0.25,
  image: 'https://test.image.jpg',
  description: 'Test product description quite long',
};

const additionalInventoryItem = {
  id: '2',
  title: 'Additional item title',
  price: 1000,
  image: 'https://additional.image.jpg',
  description: 'Shorter description',
};

const methods = {
  onInventoryItemChange: jest.fn(),
  onInventoryItemRemove: jest.fn(),
  onNewInventoryItemSubmit: jest.fn(),
};

it('renders only a new product form if items prop undefined', () => {
  const props = {};

  const { container } = render(<Inventory {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});

it('renders only a new product form if items prop is an empty array', () => {
  const props = {
    items: [],
  };

  const { container } = render(<Inventory {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});

it('renders a single product and new product form', () => {
  const props = {
    items: [initialInventoryItem],
  };

  const { container } = render(<Inventory {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});

it('renders multiple products and new product form', () => {
  const props = {
    items: [initialInventoryItem, additionalInventoryItem],
  };

  const { container } = render(<Inventory {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});
