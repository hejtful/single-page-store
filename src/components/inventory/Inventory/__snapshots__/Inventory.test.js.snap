// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a single product and new product form 1`] = `
<div>
  <div
    data-cy="inventory"
  >
    <div
      class="wrapper"
    >
      <form>
        <div
          class="formGroupInline"
        >
          <input
            class="formElement"
            placeholder="Title"
            required=""
            type="text"
            value="Test product title"
          />
          <input
            class="formElement"
            min="0.01"
            placeholder="Price"
            required=""
            step="0.01"
            type="number"
            value="0.25"
          />
        </div>
        <input
          class="formElement"
          placeholder="Image URL"
          required=""
          type="url"
          value="https://test.image.jpg"
        />
        <textarea
          class="formElement textarea"
          placeholder="Description"
          required=""
        >
          Test product description quite long
        </textarea>
        <button
          class="button danger"
          type="button"
        >
          Remove Product
        </button>
      </form>
    </div>
    <div
      class="wrapper"
    >
      <form>
        <div
          class="formGroupInline"
        >
          <input
            class="formElement"
            name="title"
            placeholder="Title"
            required=""
            type="text"
          />
          <input
            class="formElement"
            min="0.01"
            name="price"
            placeholder="Price"
            required=""
            step="0.01"
            type="number"
          />
        </div>
        <input
          class="formElement"
          name="image"
          placeholder="Image URL"
          required=""
          type="url"
        />
        <textarea
          class="formElement textarea"
          name="description"
          placeholder="Description"
          required=""
        />
        <button
          class="button"
          type="submit"
        >
          Add Product
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`renders multiple products and new product form 1`] = `
<div>
  <div
    data-cy="inventory"
  >
    <div
      class="wrapper"
    >
      <form>
        <div
          class="formGroupInline"
        >
          <input
            class="formElement"
            placeholder="Title"
            required=""
            type="text"
            value="Test product title"
          />
          <input
            class="formElement"
            min="0.01"
            placeholder="Price"
            required=""
            step="0.01"
            type="number"
            value="0.25"
          />
        </div>
        <input
          class="formElement"
          placeholder="Image URL"
          required=""
          type="url"
          value="https://test.image.jpg"
        />
        <textarea
          class="formElement textarea"
          placeholder="Description"
          required=""
        >
          Test product description quite long
        </textarea>
        <button
          class="button danger"
          type="button"
        >
          Remove Product
        </button>
      </form>
    </div>
    <div
      class="wrapper"
    >
      <form>
        <div
          class="formGroupInline"
        >
          <input
            class="formElement"
            placeholder="Title"
            required=""
            type="text"
            value="Additional item title"
          />
          <input
            class="formElement"
            min="0.01"
            placeholder="Price"
            required=""
            step="0.01"
            type="number"
            value="1000"
          />
        </div>
        <input
          class="formElement"
          placeholder="Image URL"
          required=""
          type="url"
          value="https://additional.image.jpg"
        />
        <textarea
          class="formElement textarea"
          placeholder="Description"
          required=""
        >
          Shorter description
        </textarea>
        <button
          class="button danger"
          type="button"
        >
          Remove Product
        </button>
      </form>
    </div>
    <div
      class="wrapper"
    >
      <form>
        <div
          class="formGroupInline"
        >
          <input
            class="formElement"
            name="title"
            placeholder="Title"
            required=""
            type="text"
          />
          <input
            class="formElement"
            min="0.01"
            name="price"
            placeholder="Price"
            required=""
            step="0.01"
            type="number"
          />
        </div>
        <input
          class="formElement"
          name="image"
          placeholder="Image URL"
          required=""
          type="url"
        />
        <textarea
          class="formElement textarea"
          name="description"
          placeholder="Description"
          required=""
        />
        <button
          class="button"
          type="submit"
        >
          Add Product
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`renders only a new product form if items prop is an empty array 1`] = `
<div>
  <div
    data-cy="inventory"
  >
    <div
      class="wrapper"
    >
      <form>
        <div
          class="formGroupInline"
        >
          <input
            class="formElement"
            name="title"
            placeholder="Title"
            required=""
            type="text"
          />
          <input
            class="formElement"
            min="0.01"
            name="price"
            placeholder="Price"
            required=""
            step="0.01"
            type="number"
          />
        </div>
        <input
          class="formElement"
          name="image"
          placeholder="Image URL"
          required=""
          type="url"
        />
        <textarea
          class="formElement textarea"
          name="description"
          placeholder="Description"
          required=""
        />
        <button
          class="button"
          type="submit"
        >
          Add Product
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`renders only a new product form if items prop undefined 1`] = `
<div>
  <div
    data-cy="inventory"
  >
    <div
      class="wrapper"
    >
      <form>
        <div
          class="formGroupInline"
        >
          <input
            class="formElement"
            name="title"
            placeholder="Title"
            required=""
            type="text"
          />
          <input
            class="formElement"
            min="0.01"
            name="price"
            placeholder="Price"
            required=""
            step="0.01"
            type="number"
          />
        </div>
        <input
          class="formElement"
          name="image"
          placeholder="Image URL"
          required=""
          type="url"
        />
        <textarea
          class="formElement textarea"
          name="description"
          placeholder="Description"
          required=""
        />
        <button
          class="button"
          type="submit"
        >
          Add Product
        </button>
      </form>
    </div>
  </div>
</div>
`;
